test/sequential/test-watch-mode.mjs

```diff
@@ -174,6 +175,41 @@ describe('watch mode', { concurrency: true, timeout: 60_0000 }, () => {
     });
   });
 
+  it('should watch changes to previously loaded dependencies', async () => {
+    const dependencyContent = 'module.exports = {}';
+    const dependency = createTmpFile(dependencyContent);
+    const relativeDependencyPath = `./${path.basename(dependency)}`;
+    const dependant = createTmpFile(`console.log(require('${relativeDependencyPath}'))`);
+
+    let stderr = '';
+    let stdout = '';
+    const child = spawn(execPath, ['--watch', '--no-warnings', dependant], { encoding: 'utf8' });
+    child.stdout.on('data', (data) => { stdout += data; });
+    child.stderr.on('data', (data) => { stderr += data; });
+    child.on('error', (err) => { throw err; });
+
+    await once(child.stdout, 'data');
+    rmSync(dependency, { force: true });
+
+    await setTimeout(600); // throttle + 100
+    writeFileSync(dependency, dependencyContent);
+
+    await setTimeout(600); // throttle + 100
+    child.kill();
+    await once(child, 'exit');
+
+    // TODO(ruyadorno): fs.watch is flaky when removing files from a watched
```

Comment:
Can you create an issue for this and assign it to me? And can you give platform information for this line, since fs.watch has different implementations on different OS